dataland-frontend/src/components/forms/parts/UploadReports.vue

```diff
@@ -0,0 +1,288 @@
+              :info="euTaxonomyKpiInfoMappings?.groupLevelIntegratedReport ?? 'Group Level Integrated Report'"
+              :name="'isGroupLevel'"
+            />
+          </div>
+        </FormKit>
+      </div>
+    </div>
+  </FormKit>
+</template>
+
+<script lang="ts">
+import { defineComponent } from "vue";
+
+import Calendar from "primevue/calendar";
+import UploadFormHeader from "@/components/forms/parts/UploadFormHeader.vue";
+import PrimeButton from "primevue/button";
+import FileUpload, { FileUploadEmits } from "primevue/fileupload";
+import YesNoComponent from "@/components/forms/parts/YesNoComponent.vue";
+import { formatBytesUserFriendly } from "@/utils/NumberConversionUtils";
+import { WhichSetOfFiles } from "@/components/forms/Types";
+
+export default defineComponent({
+  name: "UploadReports",
+  components: {
+    Calendar,
+    UploadFormHeader,
+    PrimeButton,
+    FileUpload,
+    YesNoComponent,
+  },
+  emits: ["selectedFiles", "removeReportFromFilesToUpload", "updateReportDateHandler"],
+  data() {
+    return {
+      formsDatesFilesToUpload: [] as string[] | undefined,
+      formatBytesUserFriendly,
+    };
+  },
+  methods: {
+    /**
+     * Function to emit event when files are selected
+     *
+     * @param event date in date format
+     * @param event.originalEvent event
+     * @param event.files files
+     */
+    onSelectedFiles(event: { files: Record<string, string>[]; originalEvent: Event }) {
+      this.$emit("selectedFiles", event);
+    },
+    /**
+     * Function to emit event when files are selected remove report from files to upload
+     *
+     * @param fileToRemove File To Remove
+     * @param fileRemoveCallback Callback function removes report from the ones selected in formKit
+     * @param index Index number of the report
+     */
+    removeReportFromFilesToUpload(
+      fileToRemove: Record<string, string>,
+      fileRemoveCallback: (x: number) => void,
+      index: number
+    ) {
+      this.$emit("removeReportFromFilesToUpload", fileToRemove, fileRemoveCallback, index);
+    },
+    /**
+     * Function to emit event to update the date of a single report file
+     *
+     * @param index file to update
+     * @param event new date value
+     * @param whichSetOfFiles which set of files will be edited
+     */
+    updateReportDateHandler(index: number, event: Date, whichSetOfFiles: WhichSetOfFiles) {
+      this.$emit("updateReportDateHandler", index, event, whichSetOfFiles);
+    },
+    /**
+     * Function to clear all not uploaded files
+     *
+     */
+    clearAllNotUploadedFiles(): void {
+      (this.$refs.fileUpload as FileUploadEmits).clear();
+    },
+  },
+
+  props: {
+    filesToUpload: {
+      type: Array,
+    },
+    uploadFiles: {
+      type: Array,
+    },
+    editMode: {
+      type: Boolean,
+    },
+    maxFileSize: {
+      type: Number,
+    },
+    euTaxonomyKpiNameMappings: {},
+    euTaxonomyKpiInfoMappings: {},
```

Comment:
I don't see no reason why this has to be a prop, since the same mapping object is always passed down here.

You could just import it inside of this component.

Another positive effect would be that the IDE won't complain anymore about these:

![image](https://user-images.githubusercontent.com/99255296/232782770-90f0d720-9632-4693-bc4d-c45a3c885c24.png)

Reply:
This component will not be used with another set of mapping variables ?

Reply:
Let's keep the props.  I think you are right and it makes sense to keep this component as generic as possible.
You could try to fix the red warnings by typing the props, right?
I mean the props have to at least contain the required fields in the DOM.

Reply:
My IDE not complaining about it.

Reply:
Ok I will investigate this further.

Reply:
Fix typing in this file!